x-pack/plugins/enterprise_search/public/applications/app_search/components/multi_input_rows/multi_input_rows_logic.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+interface MultiInputRowsValues {
+  values: string[];
+  hasEmptyValues: boolean;
+  hasOnlyOneValue: boolean;
+}
+
+interface MultiInputRowsActions {
+  addValue(): void;
+  deleteValue(indexToDelete: number): { indexToDelete: number };
+  editValue(index: number, newValueValue: string): { index: number; newValueValue: string };
+}
+
+export const MultiInputRowsLogic = kea<MakeLogicType<MultiInputRowsValues, MultiInputRowsActions>>({
```

Comment:
Does this logic need to be keyed in case there are multiple instances on a page 🤔

Reply:
oh shit, I did forget to test this with multiple value boosts on the page. hang tight..

Reply:
![Uploading latest.gif…]()

Reply:
😂 hahaaa just throw my body in the trash

Reply:
@JasonStoltz I'm going to put this down for the day because I've banged my head against it enough, but I am absolutely fucking stumped as to why keyed logic isn't working for this component:

https://github.com/elastic/kibana/pull/96881/commits/8436c812adc2283505da786e93580de936453ebc

<img width="610" alt="" src="https://user-images.githubusercontent.com/549407/114638978-9fc2f380-9c81-11eb-847e-9d13021ffe98.png">

Like. I am 100% certain I am passing correct & distinct IDs, but Kea for some reason does NOT care. I even tried the new BindLogic and nothing. I'm banging my head against the wall at this point so I'm going to set this down for today.

IMO, worst case/fallback scenario: if we can't figure this out by tomorrow, I'll convert this component back to using basic React state instead of Kea logic 🤷

Reply:
I mentioned this on Slack but leaving here for posterity: I think you need to update the `path` property of `MultiInputRowLogic` to be a function that accepts a single `key` argument:

```
  path: (key: string) => ['enterprise_search', 'app_search', 'multi_input_rows_logic', key],
```

The kea documentation [doesn't directly tell you this](https://kea.js.org/docs/guide/debugging/#setting-the-path-manually) but it specifically uses the `path` to look up things in the cache.  Normally, kea autogenerates these, and if you include a key it uses that key to generate its own internal path. But if you set the path manually AND have a key, you need to make sure you're utilizing the key in the path.

Reply:
Thanks so much y'all, I'm thrilled this is working!

Functional: https://github.com/elastic/kibana/pull/96881/commits/85b95a12c53adebb7f0267b321c47a131f7581b6
Tests: https://github.com/elastic/kibana/pull/96881/commits/a758b1f8cbe55ae922763ad7398aaf56fc239a3a

Going to bring the tests portion to the Slack channel for a discussion on whether folks are good with the API